app/lib/pdf_filler/nj1040_pdf.rb

```diff
@@ -0,0 +1,111 @@
+module PdfFiller
+  class Nj1040Pdf
+    include PdfHelper
+
+    def source_pdf_name
+      "nj1040-TY2023"
+    end
+
+    def initialize(submission)
+      @submission = submission
+
+      # Most PDF fields are grabbed right off the XML
+      builder = StateFile::StateInformationService.submission_builder_class(:nj)
+      @xml_document = builder.new(submission).document
+    end
+
+    def hash_for_pdf
+      county_code = get_county_code
+      taxpayer_ssn = get_taxpayer_ssn
+      spouse_ssn = get_spouse_ssn
+      {
+        # county code
+        "CM4": county_code[1],
+        "CM3": county_code[2],
+        "CM2": county_code[3],
+        "CM1": county_code[4],
+
+        # taxpayer ssn
+        "undefined": taxpayer_ssn[0],
+        "undefined_2": taxpayer_ssn[1],
+        "Your Social Security Number required": taxpayer_ssn[2],
+        "Text3": taxpayer_ssn[3],
+        "Text4": taxpayer_ssn[4],
+        "Text5": taxpayer_ssn[5],
+        "Text6": taxpayer_ssn[6],
+        "Text7": taxpayer_ssn[7],
+        "Text8": taxpayer_ssn[8],
+
+        # name
+        "Last Name First Name Initial Joint Filers enter first name and middle initial of each Enter spousesCU partners last name ONLY if different": get_name,
+
+        # spouse ssn
+        "undefined_3": spouse_ssn ? spouse_ssn[0] : "",
+        "undefined_4": spouse_ssn ? spouse_ssn[1] : "",
+        "undefined_5": spouse_ssn ? spouse_ssn[2] : "",
+        "Text9": spouse_ssn ? spouse_ssn[3] : "",
+        "Text10": spouse_ssn ? spouse_ssn[4] : "",
+        "Text11": spouse_ssn ? spouse_ssn[5] : "",
+        "Text12": spouse_ssn ? spouse_ssn[6] : "",
+        "Text13": spouse_ssn ? spouse_ssn[7] : "",
+        "Text14": spouse_ssn ? spouse_ssn[8] : "",
```

Comment:
[DUST] could also do something like:
```
answers = {
...county code etc...
}

if spouse_ssn
  answers.merge!({ ...spouse ssn stuff })
end
```

we have a similar pattern in other pdf fillers but def not required. interested in any opinions on one approach vs the other!